core/connectors/sources/postgres_source/src/lib.rs

```diff
@@ -0,0 +1,691 @@
+                    .map_err(|_| Error::InvalidRecord)?;
+                Ok(value
+                    .map(serde_json::Value::Bool)
+                    .unwrap_or(serde_json::Value::Null))
+            }
+            "INT2" | "INT4" | "INT8" => {
+                let value: Option<i64> = row
+                    .try_get(column_index)
+                    .map_err(|_| Error::InvalidRecord)?;
+                Ok(value
+                    .map(serde_json::Value::from)
+                    .unwrap_or(serde_json::Value::Null))
+            }
+            "FLOAT4" | "FLOAT8" | "NUMERIC" => {
+                let value: Option<f64> = row
+                    .try_get(column_index)
+                    .map_err(|_| Error::InvalidRecord)?;
+                Ok(value
+                    .map(serde_json::Value::from)
+                    .unwrap_or(serde_json::Value::Null))
+            }
+            "VARCHAR" | "TEXT" | "CHAR" => {
+                let value: Option<String> = row
+                    .try_get(column_index)
+                    .map_err(|_| Error::InvalidRecord)?;
+                Ok(value
+                    .map(serde_json::Value::String)
+                    .unwrap_or(serde_json::Value::Null))
+            }
+            "TIMESTAMP" | "TIMESTAMPTZ" => {
+                let value: Option<DateTime<Utc>> = row
+                    .try_get(column_index)
+                    .map_err(|_| Error::InvalidRecord)?;
+                Ok(value
+                    .map(|dt| serde_json::Value::String(dt.to_rfc3339()))
+                    .unwrap_or(serde_json::Value::Null))
+            }
+            "UUID" => {
+                let value: Option<Uuid> = row
+                    .try_get(column_index)
+                    .map_err(|_| Error::InvalidRecord)?;
+                Ok(value
+                    .map(|u| serde_json::Value::String(u.to_string()))
+                    .unwrap_or(serde_json::Value::Null))
+            }
+            "JSON" | "JSONB" => {
+                let value: Option<serde_json::Value> = row
+                    .try_get(column_index)
+                    .map_err(|_| Error::InvalidRecord)?;
+                Ok(value.unwrap_or(serde_json::Value::Null))
+            }
+            _ => {
+                let value: Option<String> = row
+                    .try_get(column_index)
+                    .map_err(|_| Error::InvalidRecord)?;
+                Ok(value
+                    .map(serde_json::Value::String)
+                    .unwrap_or(serde_json::Value::Null))
+            }
+        }
+    }
+
+    fn to_snake_case(&self, input: &str) -> String {
+        let mut result = String::new();
+        let mut prev_was_uppercase = false;
+
+        for (i, ch) in input.chars().enumerate() {
+            if ch.is_uppercase() {
+                if i > 0 && !prev_was_uppercase {
+                    result.push('_');
+                }
+                result.push(ch.to_lowercase().next().unwrap());
```

Comment:
Maybe let's handle this with e.g. `let Ok(ch) = ...` or change it to return `Result<String, SomeError>`, in order to avoid direct `unwrap()`?